### PR TITLE
6062 migrate user detail part 6

### PR DESCRIPTION
**Pull request by ThibaudAV**
**Issue**

https://github.com/gravitee-io/issues/issues/6062

**Description**

End migration of all userDetails features
 - reset passowrd
 - active / reject  registration 
 - add Table of content
 - add simple progress bar on each card

**Additional context**

### CODE REVIEW

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

📚 View the storybook of this branch [here](https://612657caa8e859003a8a6430-oshbdfurew.chromatic.com)

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/6062-migrate-user-detail-part-6/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

File: gravitee-apim-console-webui/src/services-ngx/users.service.spec.ts [hunk 1]
```diff
@@ -191,6 +191,21 @@ describe('UsersService', () => {
...
+      const userId = 'userId';
+
+      usersService.resetPassword(userId).subscribe(() => {
+        done();
+      });
+
+      const req = httpTestingController.expectOne(`${CONSTANTS_TESTING.org.baseURL}/users/${userId}/resetPassword`);
+      expect(req.request.method).toEqual('POST');
+
+      req.flush({});
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/services-ngx/users.service.spec.ts]:
``` suggestion
      httpTestingController.expectOne({
        method: 'POST',
        url: `${CONSTANTS_TESTING.org.baseURL}/users/${userId}/resetPassword`,
      })
      .flush({});
```

File: gravitee-apim-console-webui/src/organization/configuration/user/detail/org-settings-user-detail.component.ts [hunk 2]
```diff
@@ -241,6 +244,32 @@ export class OrgSettingsUserDetailComponent implements OnInit, OnDestroy {
...
+        },
+        role: 'alertdialog',
+        id: 'resetUserPasswordConfirmDialog',
+      })
+      .afterClosed()
+      .pipe(
+        takeUntil(this.unsubscribe$),
+        filter((confirm) => confirm === true),
+        switchMap(() => this.usersService.resetPassword(this.user.id)),
+        tap(() => this.snackBarService.success(`The password of user "${this.user.displayName}" has been successfully reset`)),
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/organization/configuration/user/detail/org-settings-user-detail.component.ts:264]: catchError + error snackbar?

File: gravitee-apim-console-webui/src/organization/configuration/user/detail/org-settings-user-detail.component.ts [hunk 3]
```diff
@@ -241,6 +244,32 @@ export class OrgSettingsUserDetailComponent implements OnInit, OnDestroy {
...
+  onRestPassword() {
+    this.matDialog
+      .open<GioConfirmDialogComponent, GioConfirmDialogData>(GioConfirmDialogComponent, {
+        width: '500px',
+        data: {
+          title: 'Reset user password',
+          content: `
+          Are you sure you want to reset password of user <strong>${this.user.displayName}</strong> ?
+          <br>
+          An email with a link to change it will be sent to him.
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/organization/configuration/user/detail/org-settings-user-detail.component.ts]: Maybe we need to check with tech writers about the `to him`

File: gravitee-apim-console-webui/src/services-ngx/group.service.spec.ts [hunk 4]
```diff
@@ -80,6 +80,22 @@ describe('GroupService', () => {
...
+      const memberId = 'memberId';
+
+      groupService.deleteMember(groupId, memberId).subscribe(() => {
+        done();
+      });
+
+      const req = httpTestingController.expectOne(`${CONSTANTS_TESTING.env.baseURL}/configuration/groups/${groupId}/members/${memberId}`);
+      expect(req.request.method).toEqual('DELETE');
+
+      req.flush({});
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/services-ngx/group.service.spec.ts]:
```suggestion
      const req = httpTestingController.expectOne({
        method: 'DELETE',
        url: `${CONSTANTS_TESTING.env.baseURL}/configuration/groups/${groupId}/members/${memberId}`,
      }).flush({});
```

**Review by gaetanmaisse — APPROVED** [review]
Regarding `Progress bar on bottom of each card` I think it's a good idea but with a fast network it makes the screen flickers which is not super great for users. To discuss with Greg

File: gravitee-apim-console-webui/src/services-ngx/users.service.spec.ts (hunk 1 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/services-ngx/users.service.spec.ts]: haha more line like this ^^ but ok as you wish

File: gravitee-apim-console-webui/src/organization/configuration/user/detail/org-settings-user-detail.component.ts (hunk 2 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/organization/configuration/user/detail/org-settings-user-detail.component.ts:264]: year same for registration and groups

File: gravitee-apim-console-webui/src/organization/configuration/user/detail/org-settings-user-detail.component.ts (hunk 3 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/organization/configuration/user/detail/org-settings-user-detail.component.ts]: you doubt about the feature ou about english ? 

maybe i could change to something like this : 
`The user will receive an email with a link to set new password.`

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

📚 View the storybook of this branch [here](https://612657caa8e859003a8a6430-blyebfnyuq.chromatic.com)

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/6062-migrate-user-detail-part-6/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

File: gravitee-apim-console-webui/src/organization/configuration/user/detail/org-settings-user-detail.component.ts (hunk 3 above)
Comment (gaetanmaisse) [reply gravitee-apim-console-webui/src/organization/configuration/user/detail/org-settings-user-detail.component.ts]: `The user will receive an email with a link to set a new password.`

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

📚 View the storybook of this branch [here](https://612657caa8e859003a8a6430-woqtgqprjc.chromatic.com)

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/6062-migrate-user-detail-part-6/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

**Pull request merged**